doc/design/python_api.md

```diff
@@ -0,0 +1,156 @@
+# Design Doc: Python API
+
+The top level user API in Python should be as same as API in `paddle.v2` after refactoring Paddle from a layer based framework to an operator based framework. There are many new classes in C++ in [compile time] for describing neural networks, such as `Variable`, `Operator`, `Block`. The issue about current design is how to give a proper way to wrap the C++ API to `paddle.v2` API and writing layers in Python.
+
+This implementation of Python API includes two steps.
+
+1. Implement the Python API using current C++ runtime concepts.
+2. Replace the implementation by using compile-time concepts when they are completed.
+
+The implementation of the first step is a temporary implementation. We should design our Python API concepts based on `compile-time` concepts. We just use `runtime` classes to implement it for now.
+
+
+## Python Class and compile-time protobuf
+
+Since we design our Python API concepts based on `compile-time`, we try to map our Python classes to every compile-time result, i.e., the protobuf messages. They are:
```

Comment:
every compile-time result -> each compile-time concepts?

Reply:
It is the result I want to say, not concepts. The result of compile-time is protobuf.